Helper/OmiseMoney.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Omise\Payment\Helper;
+
+class OmiseMoney
+{
+
+    private $zeroDecimalCurrencies = ['JPY'];
+
+    /**
+     * @var integer|float
+     */
+    private $amount;
+
+    /**
+     * @var string
+     */
+    private $currency;
+
+    /**
+     * @param integer|float $amount
+     * @param this
+     */
+    public function parse($amount, $currency)
```

Comment:
Could we please rename `parse` to some other name? `parse` is usually used for methods that take in strings and parse them into other data types. But here we are just setting values. So, maybe `setAmountAndCurrency` will be more descriptive or any other name you have in mind but not `parse`.

Reply:
can we go with `init` or `setValues`?

Reply:
Is this not a constructor? https://www.php.net/manual/en/language.oop5.decon.php

Reply:
Yes it can be constructor, but I am trying to avoid creating new class every times when we want to convert units.

Reply:
It just because I can inject dependency into other dependency. Magento design is mostly dependency injection pattern.

example here.
```
 public function __construct(
        Helper $helper,
        ReturnUrlHelper $returnUrl,
        Config $config,
        Capabilities $capabilities,
        OmiseMoney $money
    ) {
        $this->helper = $helper;
        $this->returnUrl = $returnUrl;
        $this->config = $config;
        $this->capabilities = $capabilities;
        $this->money = $money;
    }
```

Reply:
Having no constructor does not help in unnecessary memory allocation. PHP will allocate memory for the object regardless whether the class has a constructor or not.

Personally, I would avoid setting values in constructor because the dependency is injected by the framework/platform via some configuration code and it will be difficult to change since the values would be dynamic. I would prefer setter method instead.

```
$this->money->setMoney($item['base_original_price']);
$this->money->setCurrency($currency);
$this->money->toSubunit();

// OR

$this->money->setMoneyAndCurrency($item['base_original_price'], $currency);
$this->money->toSubunit();

// OR
$this->money->setValues([
    'amount' => $item['base_original_price']],
    'currency' => $currency
]);
$this->money->toSubunit();
```

If we want to use `setValues`, i would prefer an array type so that it would be more clear what values we are setting.